src/core/symbology/qgsmarkersymbollayer.cpp

```diff
@@ -2325,7 +2330,8 @@ void QgsSvgMarkerSymbolLayer::renderPoint( QPointF point, QgsSymbolRenderContext
 
   bool hasDataDefinedSize = false;
   const double scaledWidth = calculateSize( context, hasDataDefinedSize );
-  const double width = context.renderContext().convertToPainterUnits( scaledWidth, mSizeUnit, mSizeMapUnitScale );
+  const double devicePixelRatio = context.renderContext().devicePixelRatio();
+  const double width = context.renderContext().convertToPainterUnits( scaledWidth, mSizeUnit, mSizeMapUnitScale ) * devicePixelRatio;
```

Comment:
@nyalldawson , I've tested the PR, and this line is problematic as it doubles the size of SVG markers when rendering from a non-cached QPicture. The width should only be doubled when rendering from the cached QImage.

Reply:
Should be fixed now